vpd-tool/src/vpd_tool.cpp

```diff
@@ -1,20 +1,76 @@
 #include "vpd_tool.hpp"
 
+#include "tool_constants.hpp"
+#include "tool_types.hpp"
+#include "tool_utils.hpp"
+
+#include <iostream>
+
 namespace vpd
 {
-int VpdTool::readKeyword(const std::string& i_fruPath,
+int VpdTool::readKeyword(const std::string& i_vpdPath,
                          const std::string& i_recordName,
                          const std::string& i_keywordName,
                          const bool i_onHardware,
                          const std::string& i_fileToSave)
 {
-    // ToDo: Need to add implementation details
-    (void)i_fruPath;
-    (void)i_recordName;
-    (void)i_keywordName;
-    (void)i_onHardware;
-    (void)i_fileToSave;
-
-    return 0;
+    int l_rc = -1;
+    try
+    {
+        types::DbusVariantType l_keywordValue;
+        if (i_onHardware)
+        {
+            // TODO: Implement read keyword's value from hardware
+        }
+        else
+        {
+            std::string l_inventoryObjectPath(constants::baseInventoryPath +
+                                              i_vpdPath);
+
+            l_keywordValue = utils::readDbusProperty(
+                constants::inventoryManagerService, l_inventoryObjectPath,
+                constants::ipzVpdInfPrefix + i_recordName, i_keywordName);
+        }
+
+        if (const auto l_value =
+                std::get_if<types::BinaryVector>(&l_keywordValue);
+            l_value && !l_value->empty())
+        {
+            // ToDo: Print value in both ASCII and hex formats
+            const std::string& l_keywordStrValue =
+                utils::getPrintableValue(*l_value);
+
+            if (i_fileToSave.empty())
+            {
+                nlohmann::json l_resultInJson = nlohmann::json::object({});
+                nlohmann::json l_keywordValInJson = nlohmann::json::object({});
+
+                l_keywordValInJson.emplace(i_keywordName, l_keywordStrValue);
+                l_resultInJson.emplace(i_vpdPath, l_keywordValInJson);
+
+                utils::printJson(l_resultInJson);
+            }
+            else
+            {
+                // TODO: Write result to a given file path.
+            }
+            l_rc = 0;
```

Comment:
Just to send return code as success/failure from one common place this initialization has been done here?
At first sight it looks like a random magic number assignment. In case you want to have these kind of assignments, at least have a constant defined to depict its actual meaning. Here "0" does not mean Zero it means success.

Reply:
Add a TODO and create a defect for now, we will have to take it up at repo level.